Test Plan Automations Examples/Update_Testplans_With_Template_Sample_Script.ipynb

```
@@ -0,0 +1,417 @@
+    "  },\n",
+    "  \"systemlink\": {\n",
+    "    \"parameters\": [\n",
+    "        {\n",
+    "            \"display_name\": \"Test plan IDs\",\n",
+    "            \"id\": \"test_plan_ids\",\n",
+    "            \"type\": \"string[]\"\n",
+    "        },\n",
+    "        {\n",
+    "            \"display_name\": \"Test plan template ID\",\n",
+    "            \"id\": \"template_id\",\n",
+    "            \"type\": \"string\"\n",
+    "        }\n",
+    "    ],\n",
+    "    \"version\": 1\n",
+    "  },\n",
+    "  \"tags\": [\"parameters\"]\n",
+    "}\n",
+    "```\n",
+    "For more information on how parameterization works, review the [papermill documentation](https://papermill.readthedocs.io/en/latest/usage-parameterize.html#how-parameters-work)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7af836c5-4a7f-46b4-81e6-ecfb72580822",
+   "metadata": {
+    "editable": true,
+    "papermill": {
+     "parameters": {
+      "template_id": "",
+      "test_plan_ids": []
+     }
+    },
+    "slideshow": {
+     "slide_type": ""
+    },
+    "systemlink": {
+     "outputs": [
+      {
+       "display_name": "Total test plans",
+       "id": "total_test_plans",
+       "type": "scalar"
+      },
+      {
+       "display_name": "Test plans updated",
+       "id": "test_plans_updated",
+       "type": "string[]"
+      },
+      {
+       "display_name": "Test plans not updated",
+       "id": "test_plans_not_updated",
+       "type": "string[]"
+      }
+     ],
+     "parameters": [
+      {
+       "display_name": "Test plan IDs",
+       "id": "test_plan_ids",
+       "type": "string[]"
+      },
+      {
+       "display_name": "Test plan template ID",
+       "id": "template_id",
+       "type": "string"
+      }
+     ],
+     "version": 1
+    },
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "test_plan_ids = [] #Add the test plans ids as list of strings here\n",
```

Are we not validating if this array is empty before performing any operations, I can see you validate for template_id but not this list. We should do that and fail fast.

---

Added a check to verify that